newrelic/resource_newrelic_one_dashboard.go

```diff
@@ -0,0 +1,380 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/newrelic/newrelic-client-go/pkg/entities"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+)
+
+func resourceNewRelicOneDashboard() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNewRelicOneDashboardCreate,
+		Read:   resourceNewRelicOneDashboardRead,
+		Update: resourceNewRelicOneDashboardUpdate,
+		Delete: resourceNewRelicOneDashboardDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The dashboard's name.",
+			},
+			"page": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "",
+				Elem:        dashboardPageSchemaElem(),
+			},
+			// Optional
+			"account_id": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "The New Relic account ID where you want to create the dashboard.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The dashboard's description.",
+			},
+			"permissions": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "public_read_only",
+				ValidateFunc: validation.StringInSlice([]string{"private", "public_read_only", "public_read_write"}, false),
```

Comment:
Not a blocker, but we've usually tried to make things case-insensitive for constants like these (per JT suggestions 😉 ), but I also see the merit with maintaining a slightly stricter API.

Reply:
They must be UPPERCASE for the API, and are upcase/tolower in the expand/flatten funcs, so updating the validator to not care is fine with me.